README.md

```diff
@@ -0,0 +1,566 @@
+MAY *NOT* be completely accurate according to slippage or other on-chain conditions,
+when performing the actual exchange.
+
+In certain types of fee calculations, this calculation MAY *NOT* reflect the "per-user" price-per-share,
+and instead should reflect the "average-user's" price-per-share,
+meaning what the average user should expect to see when exchanging to and from.
+The `assetsOf` method SHOULD be used for more accurate calculations of a user's underlying balance.
+
+
+```yaml
+- name: assetsPerShare
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+  - name: assetsPerUnitShare
+    type: uint256
+```
+
+#### assetsOf
+
+Total number of underlying assets that `depositor`'s shares represent.
+
+MAY be more accurate than using `assetsPerShare` or `totalAssets / Vault.totalSupply`
+for certain types of fee calculations.
+MAY *NOT* be completely accurate according to slippage or other on-chain conditions.
+
+```yaml
+- name: assetsOf
+  type: function
+  stateMutability: view
+
+  inputs:
+  - name: depositor
+    type: address
+
+  outputs:
+  - name: assets
+    type: uint256
+```
+
+#### maxDeposit
+
+Total number of underlying assets that can be deposited.
+
+MUST return `2 ** 256 - 1` to specify there is no maximum amount of assets that may be deposited.
+MAY be used in the `previewDeposit` or `deposit` methods for `assets` input parameter.
+
+```yaml
+- name: maxDeposit
+  type: function
+  stateMutability: view
+
+  inputs: []
+
+  outputs:
+  - name: maxAssets
+    type: uint256
+```
+
+#### previewDeposit
+
+Allows an on-chain or off-chain user to simulate the effects of their deposit at the current block,
+given current on-chain conditions.
+
+MUST return the *exact* amount of Vault shares that would be obtained if depositing a given
+*exact* amount of underlying assets using the `deposit` method.
+MUST be inclusive of deposit fees. Integrators should be aware of the existance of deposit fees.
+
+Note that any discrepency between `shares * pricePerShare` and `assets` SHOULD be considered slippage
+in share price or some other type of condition, meaning the depositor will lose assets by depositing.
+
+```yaml
+- name: previewDeposit
+  type: function
+  stateMutability: view
+
+  inputs:
+  - name: assets
+    type: uint256
+
+  outputs:
+  - name: shares
+    type: uint256
+```
+
+#### deposit
+
+Mints `shares` Vault shares to `receiver` by depositing exactly `amount` of underlying tokens.
+
+MUST match exactly the quote given by `previewDeposit` executed immediately before `deposit` in the same transaction.
+Any discrepency could cause a revert due to tight slippage bounds by caller.
+MUST emit the `Deposit` event.
+MAY support an additional flow in which the underlying tokens are owned by the Vault contract
+before the `deposit` execution, and are accounted for during `deposit`.
+SHOULD add a means to revert due to slippage loss if the implementation intends to support EOA account access directly.
```

Comment:
```suggestion
```

Reply:
Can we change this to more of a general note and keep it? Agree with not `SHOULD`

Reply:
Can I see what you had in mind?

Reply:
```suggestion

If implementors intend to support EOA account access directly, they should consider adding an additional function with the means to revert due to slippage loss since they are unable to revert if the output is undesired.
```